akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/scaladsl/ReplicatedEventSourcingOverGrpc.scala

```diff
@@ -0,0 +1,180 @@
+        ReplicatedEventSourcing.externalReplication(replicationId, allReplicaIds)(replicatedBehaviorFactory)
+      })
+
+    val sharding = ClusterSharding(system)
+    sharding.init(replicatedEntity.entity)
+
+    // sharded daemon process for consuming event stream from the other dc:s
+    settings.otherReplicas.foreach { otherReplica =>
+      startConsumer(
+        settings.entityTypeKey.name,
+        settings.selfReplicaId,
+        otherReplica,
+        settings.protobufDescriptors,
+        sharding.entityRefFor(replicatedEntity.entity.typeKey, _))
+    }
+
+    // FIXME user still has to start the producer, could we opt-out do that as well?
+    new ReplicatedEventSourcingOverGrpc[Command](eventProducerRoute, replicatedEntity.entity.typeKey)
+  }
+
+  private def startConsumer[C](
+      entityTypeKeyName: String,
+      selfReplicaId: ReplicaId,
+      remoteReplica: Replica,
+      protobufDescriptors: Seq[Descriptors.FileDescriptor],
+      entityRefFactory: String => EntityRef[C])(implicit system: ActorSystem[_]): Unit = {
+    implicit val timeout: Timeout = 5.seconds // FIXME from config
+    implicit val ec: ExecutionContext = system.executionContext
+
+    val projectionName = s"${entityTypeKeyName}_${remoteReplica.replicaId.id}"
+    val sliceRanges = Persistence(system).sliceRanges(remoteReplica.numberOfConsumers)
+
+    val eventsBySlicesQuery =
+      GrpcReadJournal(remoteReplica.grpcQuerySettings, remoteReplica.grpcClientSettings, protobufDescriptors)
+
+    ShardedDaemonProcess(system).init(projectionName, remoteReplica.numberOfConsumers, { idx =>
+      val sliceRange = sliceRanges(idx)
+      val projectionKey =
+        s"${eventsBySlicesQuery.streamId}-${remoteReplica.replicaId.id}-${sliceRange.min}-${sliceRange.max}"
+      val projectionId = ProjectionId.of(projectionName, projectionKey)
+
+      val replicationFlow = FlowWithContext[EventEnvelope[_], ProjectionContext].mapAsync(1) {
+        envelope =>
+
+          val replicatedEventMetadata = envelope.eventMetadata match {
+            case Some(rm: ReplicatedEventMetadata) => rm
+            case other                             => throw new IllegalArgumentException(s"Unknown type or missing metadata: [$other]")
+          }
+          if (replicatedEventMetadata.originReplica == remoteReplica.replicaId) {
+            val replicationId = ReplicationId.fromString(envelope.persistenceId)
+            val destinationReplicaId = replicationId.withReplica(selfReplicaId)
+            val entityRef = entityRefFactory(destinationReplicaId.entityId).asInstanceOf[EntityRef[PublishedEvent]]
+            log.info(
+              "[{}], Forwarding event originating on dc [{}] to [{}] (version: [{}]): [{}]",
+              system.name,
+              replicatedEventMetadata.originReplica,
+              destinationReplicaId.persistenceId.id,
+              replicatedEventMetadata.version,
+              envelope.event)
+            val askResult = entityRef.ask[Done](
```

Comment:
It will be tempting to increase the parallelism of the mapAsync, but then we must be careful. Messages can be lost from here to the destination entity and we accept higher sequence numbers and can't validate exact sequence number. Then the following is possible:
1. send e1, e2, e3 to the entity
2. e1 arrives all fine
3. e2 is lost
4. e3 arrives and is accepted since it has a higher sequence number
5. there is a timeout of the ack for e2 and the stream is restarted
6. e2 is sent again, arrives but is handled as s duplicate because it has a lower sequence number than already handled e3

Once again we would need something like the mapAsyncPartitioned.

Reply:
Lightweight internal mapAsyncPartitioned operator added in ef8f50d3621bb65150df1c7bbe6584e1431e3cf5